callstats-java-sdk/src/main/java/io/callstats/sdk/CallStats.java

```
@@ -322,7 +333,58 @@ public void onFailure(Exception e) {
+		}
+		List<ConferenceStats> tempStats = confereneStatsMap.get(userID);
+		if (tempStats == null) {
+			tempStats = new ArrayList<ConferenceStats>();
+			confereneStatsMap.put(userID, tempStats);
+		}
+	}
+
+	public synchronized void stopStatsReportingForUser(String userID) {
+		if (userID == null) {
+			logger.error("stopStatsReportingForUser: Arguments cannot be null ");
+			throw new IllegalArgumentException("stopStatsReportingForUser: Arguments cannot be null");
+		}
+		List<ConferenceStats> tempStats = confereneStatsMap.get(userID);
+
+		if (tempStats != null && tempStats.size() > 0) {
+			StreamStats streamStats;
+			StreamStatsData streamStatsData;
+			ConferenceStats conferenceStats = tempStats.get(0);
+			ConferenceStatsData conferenceStatsData = new ConferenceStatsData(conferenceStats.getLocalUserID());
+			UserInfo info = new UserInfo(conferenceStats.getConfID(), conferenceStats.getLocalUserID(), conferenceStats.getUcID());
+			for (ConferenceStats stats : tempStats) {
+				streamStatsData = new StreamStatsData(stats.getRtt(), stats.getPacketsSent(), stats.getBytesSent(), stats.getJitter());
+				streamStats = new StreamStats(stats.getFromUserID(), stats.getToUserID(), stats.getStatsType(), streamStatsData);
+				conferenceStatsData.addStreamStats(stats.getSsrc(), streamStats);
+			}
+
+			String statsString = gson.toJson(conferenceStatsData);
+			logger.debug("Stats string is " + statsString);
+			sendCallStatsConferenceStats(statsString, info);
+		}
+	}
+
+	public synchronized void sendConferenceStats(String userID, ConferenceStats stats) {
+		if (stats == null || userID == null) {
+			logger.error("sendConferenceStats: Arguments cannot be null ");
+			throw new IllegalArgumentException("sendConferenceStats: Arguments cannot be null");
+		}
+		List<ConferenceStats> tempStats = confereneStatsMap.get(userID);
+		if (tempStats == null) {
+			tempStats = new ArrayList<ConferenceStats>();
```

if tempStats is null, it means that startReportingForUser() wasn't called. Should we throw an exception here?